overlord/hookstate/ctlcmd/ctlcmd.go

```diff
@@ -66,6 +68,13 @@ func (c *baseCommand) context() *hookstate.Context {
 	return c.c
 }
 
+func (c *baseCommand) ensureContext() (context *hookstate.Context, err error) {
+	if c.c == nil {
+		err = errors.New(i18n.G(`The "snapctl" command can only be invoked from within a snap`))
```

Comment:
Just a quick drive-by highlevel comment - I think we should not have go error types localized. Instead I would suggest the pattern is something like:
```
var ErrNoContext = errors.New("cannot use snapctl without a context")

 func (c *baseCommand) ensureContext() (context *hookstate.Context, err error) {
    ...
    return ErrNoContext
}
```
and then in the relevant "main" (cmd/snapctl/main.go I presume) we check for ErrNoContext and if we see that we print the localized error and exit. 

I hope this makes sense, sorry that it's a bit terse, I'm a bit short on time right now.

Reply:
Mmm... I do recognize the problem, but I think that a proper solution cannot be done in the context of this PR. The problem is far from trivial: we have a process boundary between snapctl (which knows the user's locale) and snapd (which doesn't). There are (at least) two possible approaches:

1. Have snapd return errors in machine form, either as an untranslated string (as you suggest), or as error codes. Then the client will translate them properly when presenting them to the user.
2. Have the client pass locale information to snapd over the REST API, and make `i18n.G()` (or some other function) aware of the context so that it would translate the messages to the locale encoded in the context.

The first option has the big downside that it would only work for fixed messages (a `%s` in them would immediately make the message non translatable), unless we design a way to pass more context variables back to the client process and manage to build a sensible error message there.

Anyway, regardless of the path we choose, both require a substantial amount of work, as both snapctl and snapd are not ready for either solution. That's why I propose to leave this for later. :-)

Reply:
Indeed this is more complicated due to how snapctl is implemented, we already have similar case with ctlcmd.UnsuccessfulError type and respective client.ErrorKindUnsuccessful which requires changes in api_snapctl.go, cmd/snapctl/main.go (as well as overlord/hookstate/ctlcmd/..); I'm not sure it's worth doing atm, but if we want it localized, then we should follow this pattern.

But I agree with the high level remark from @mvo5, the error itself shouldn't be localized.

Reply:
and nitpick^2, the error isn't strictly true because we deliberately allow 'snapctl --help' invocation from the shell; I think the error message I suggested below in the other comment would solve it.

Reply:
OK, I removed the i18n call. However, there are still many of such calls in this module; whould I remove all of them (maybe in a separate PR)?